### PR TITLE
Update a bunch of libraries

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Update Gradle 7.1.1 -> 7.3.1 using git bash ([changelog](https://docs.gradle.org/7.3.1/userguide/upgrading_version_7.html))
- Update google-auto-service 1.0 -> 1.0.1 ([changelog](https://github.com/google/auto/releases/tag/auto-service-1.0.1))
- Update Mockito 3.6.0 -> 4.0.0 ([changelog](https://github.com/mockito/mockito/releases))
- Update ConstraintLayout 2.0.4 -> 2.1.2 ([changelog](https://developer.android.com/jetpack/androidx/releases/constraintlayout))
- Update Google Material 1.2.1 -> 1.4.0 ([changelog](https://github.com/material-components/material-components-android/releases))
- Update Jsoup 1.13.1 -> 1.14.3 ([changelog](https://github.com/jhy/jsoup/blob/master/CHANGES))
- Update RxJava3 3.0.7 -> 3.0.13 ([changelog](https://github.com/ReactiveX/RxJava/releases))
- Update PrettyTime 5.0.1.Final -> 5.0.2.Final ([changelog](https://github.com/ocpsoft/prettytime/compare/5.0.1.Final...5.0.2.Final))
- Update AndroidX Test Junit 1.1.2 -> 1.1.3 ([changelog](https://developer.android.com/jetpack/androidx/releases/test))
- Remove unused Espresso testing library

#### APK testing 
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-all.zip
-distributionSha256Sum=9bb8bc05f562f2d42bdf1ba8db62f6b6fa1c3bf6c392228802cc7cb0578fe7e0
```
Comment (XiangRongLin) [inline gradle/wrapper/gradle-wrapper.properties:4]: The hash is automatically changed or manually? If manually from where is it?

**Review by XiangRongLin — COMMENTED** [review]
The build is failing.

---

Did you check for API changes?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (TacoTheDank) [reply gradle/wrapper/gradle-wrapper.properties:4]: Changed manually. I got it from https://services.gradle.org/distributions/.

**Comment by litetex** [comment]
Downgraded to draft as build is failing.

**Comment by TacoTheDank** [comment]
Kotlin 1.6 was changing how AndroidX Room was working, see here: https://developer.android.com/jetpack/androidx/releases/room#2.4.0-rc01. Should wait for AndroidX Room 2.4.0 before upgrading to Kotlin 1.6.

**Comment by TacoTheDank** [comment]
Added runners library non-transitively to resolve `androidx.test.filters.LargeTest` in ErrorInfoTest.

**Comment by TacoTheDank** [comment]
> Did you check for API changes?

Seems I somehow missed this: yes :) everything looks good, I didn't see anything in the changelogs that was noteworthy or that required code changes.

**Comment by litetex** [comment]
@TacoTheDank 
I noticed that the APK is now ~600kB bigger than before. Do you know where that comes from?

**Review by litetex — COMMENTED** [review]
Did a quick test, LGTM.

If my comment above is clarified the PR get's my approval

**Comment by TacoTheDank** [comment]
> @TacoTheDank I noticed that the APK is now ~600kB bigger than before. Do you know where that comes from?

So I analyzed "before" and "after" APKs...

Results: Google Material resources be getting thicc 😔 

Also, I bumped AGP to 7.0.4 because it [just got released](https://androidstudio.googleblog.com/2021/12/android-studio-arctic-fox-202031-patch.html).

**Review by litetex — APPROVED** [review]
LGTM

**Pull request merged**